src/offsetPath.js

```diff
@@ -18,12 +18,30 @@
       return undefined;
     } else if (ray !== null) {
       inputType = 'ray';
-      var rayInput = ray[1];
-      var rayInputDegrees = parseAngleAsDegrees(rayInput);
-      if (rayInputDegrees === null) {
+      var rayInput = ray[1].split(/\s+/);
+      if (rayInput.length > 3) {
         return undefined;
       }
-      return {type: inputType, input: rayInputDegrees};
+
+      var result = {type: inputType, input: 'none', contain: false, size: 'none'};
+      var validSizes = ['closest-side', 'farthest-side', 'closest-corner', 'farthest-corner'];
+
+      for (var i = 0; i < rayInput.length; i++) {
+        if (rayInput[i] === 'contain' && !result.contain) {
+          result.contain = true;
+        } else if (result.size === 'none' && isInArray(validSizes, rayInput[i])) {
```

Comment:
This would become something like

else if (isInArray(validSizes, rayInput[i])) {
if (result.size !== 'none') return ...;
}